### PR TITLE
New pausing

### PR DESCRIPTION
**Pull request by AbdullahDab01**

### CODE REVIEW

**Review by rojtekin — APPROVED** [review]
lgtm

File: core/src/application/Game.java [hunk 1]
```diff
@@ -20,14 +23,28 @@ public void create () {
...
+		if(Gdx.input.isKeyPressed(Input.Keys.ESCAPE)) {
+			pauseGame();
+		}
+		if (!gamePaused) {
+			controller.update();
+			model.update();
+			Time.getInstance().tick();
+		}
+		view.update(gamePaused);
 	}
```
Comment (Suedie) [inline core/src/application/Game.java:40]: Should Override pause() from ApplicationAdapter and call pauseGame().

File: core/src/view/HUD.java [hunk 2]
```diff
@@ -58,13 +58,22 @@ public class HUD {
...
     private final Label scoreLabel = new Label("Score: ", whiteTextColorAndFont);
     private final Label perkLabel = new Label("Perk: ", whiteTextColorAndFont);
     private final Label experienceLabel = new Label("XP: ", whiteTextColorAndFont);
     private final Label levelLabel = new Label("Level: ", whiteTextColorAndFont);
 
+    private final Label pauseLabel = new Label("THE GAME IS PAUSED" , new Label.LabelStyle((new BitmapFont()), Color.WHITE));
+    private final Label pauseInfoLabel = new Label("To Resume, Please Press ESC" , new Label.LabelStyle((new BitmapFont()), Color.WHITE));
+    private final Label howToPlayLabel = new Label("How to play:" , new Label.LabelStyle((new BitmapFont()), Color.WHITE));
+     private final Label toMoveLabel = new Label("To move: use arrows or W,A,S,D" , new Label.LabelStyle((new BitmapFont()), Color.WHITE));
+     private final Label toAttackLabel = new Label("To Attack: use I,J,K,L" , new Label.LabelStyle((new BitmapFont()), Color.WHITE));
```
Comment (Suedie) [inline core/src/view/HUD.java]: Extra space looks weird

File: core/src/view/View.java [hunk 3]
```diff
@@ -124,10 +130,14 @@ public void dispose () {
...
-     * Otherwise, the frame is set to the first frame of the walk animation.
-     */
+
+
+
+
+        /**
+         * Changes the player walk frame to the next frame in the walk animation after a certain time interval if the player is moving.
+         * Otherwise, the frame is set to the first frame of the walk animation.
+         */
```
Comment (Suedie) [inline core/src/view/View.java]: Why was this extra indented?

**Review by Suedie — APPROVED** [review]
lgtm

**Pull request merged**